cpp/src/arrow/vendored/datetime/tz.cpp

```diff
@@ -202,7 +202,12 @@ get_known_folder(const GUID& folderid)
     if (SUCCEEDED(hr))
     {
         co_task_mem_ptr folder_ptr(pfolder);
-        folder = std::string(folder_ptr.get(), folder_ptr.get() + wcslen(folder_ptr.get()));
+        folder.clear();
```

Comment:
An alternate workaround would be to `#pragma warning(suppress: 4244) // narrowing wchar_t -> char` here.